src/error_reporter.rs

```diff
@@ -241,6 +241,91 @@
 //!     println!("{}", report);
 //! }
 //! ```
+//!
+//! It also is not required that the wrapped error be `'static`. However, if the inner reference
+//! doesn't itself implement `Error`, then it can't be printed by `std::Report` via
+//! `Error::source`. In this situation, use `Display` instead:
```

Comment:
I don't think this second line belongs here since it's more or less just an explanation of how to implement the `Error` trait. We should assume users know that they should be using `source` for source errors that impl the error trait, and any other context they carry in their type should either be printed in the error message (`Display`) or should be returned via generic member access once that is available.

```suggestion
//! It also is not required that the wrapped error be `'static`.
```